### PR TITLE
feature(documentation-website): add language switch on footer (#702)

### LINKED ISSUE #702: Add language switch to website
## Is your feature request related to a problem? Please describe

It is currently very hard to test translations.

## Describe the solution you'd like

Add a seleect to the  footer, that switches thee desired language and stores the information in the localStorage, so that you still havee that choice upon reloading.

## Additional context

Prefill it with the currently used browser language until something else is chosen.

### PR DESCRIPTION
**Pull request by marikadeveloper**
add language switch as a html select inside the footer

Closes #702

# The Pull Request is ready

- [ ] fixes #702 
- [ ] all actions are passing
- [ ] only fixes a single issue

## Overview

This pull request introduces a html select in the footer of the website, created inside the LanguageSwitch component, that lets the user change language, selecting form the available languages. The default language is the language of the client's browser

## Review points

- should the language selected be in uppercase?
- should we add an icon like a globe icon?
- should we add the countries flags?
- should tests be added?

## Documentation-Website

- [ ] mobile view is usable
- [ ] desktop view is usable
- [ ] no a-tags are used directly (NavLink, MailLink, ExternalLink instead)
- [ ] all new texts are added to the translation files (at least the english one)
- [ ] tests have been added (if required)
- [ ] shared code has been extracted in a different file

### CODE REVIEW

File: documentation-website/src/components/language-switch.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import React, {
+  lazy, Suspense,
```
Comment (Idrinth) [inline documentation-website/src/components/language-switch.tsx:2]: formatting ;)

File: documentation-website/src/components/language-switch.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    return {
+      default: () => <select
+        className='language-switch'
+        aria-label={ ariaLabel }
+        value={ language }
+        onChange={ (event,) => changeLanguage(event.target.value,) }
+      >{ languages.map((lang,) => <option
+          key={ lang }
+          value={ lang }
+        >{ lang }</option>,) }</select>,
```
Comment (Idrinth) [inline documentation-website/src/components/language-switch.tsx:35]: maybe refer to
```tsx
<Lang lnkey={`languages.${ lang }` as languageKey}/>
```
here?

**Review by Idrinth — CHANGES_REQUESTED** [review]
Looks very good!

File: documentation-website/src/components/language-switch.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  const [
+    language,
+    setLanguage,
+  ] = useState<string>(() => that?.localStorage?.getItem('language',) ?? 'en',);
+
+  const changeLanguage = (newLanguage: string,) => {
+    setLanguage(newLanguage,);
+    that?.localStorage?.setItem('language', newLanguage,);
+    // reload page
+    that?.location?.reload();
```
[the remark this answers is not in the document]
Comment (marikadeveloper) [reply documentation-website/src/components/language-switch.tsx:21]: @Idrinth is this to be included in the testing task? (for reference: https://github.com/Idrinth/api-bench/pull/874#discussion_r1576154711)

File: documentation-website/src/components/language-switch.scss [hunk 4]
```diff
@@ -0,0 +1,8 @@
+select.language-switch {
+  appearance: auto;
+  cursor: pointer;
+  &:focus,
+  &:focus-visible {
+
+  }
```
[the remark this answers is not in the document]
Comment (marikadeveloper) [reply documentation-website/src/components/language-switch.scss]: I did this in a recent commit

File: documentation-website/src/components/language-switch.scss [hunk 5]
```diff
@@ -0,0 +1,8 @@
+select.language-switch {
+  appearance: auto;
+  cursor: pointer;
+  &:focus,
```
[the remark this answers is not in the document]
Comment (marikadeveloper) [reply documentation-website/src/components/language-switch.scss]: This has been removed in a recent commit

File: documentation-website/src/components/language-switch.tsx (hunk 2 above)
Comment (marikadeveloper) [reply documentation-website/src/components/language-switch.tsx:35]: Screenshot 2024-04-23 at 16 10 48
is it good like this?

Comment (Idrinth) [reply documentation-website/src/components/language-switch.tsx]: perfect!

File: documentation-website/src/components/language-switch.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (Idrinth) [reply documentation-website/src/components/language-switch.tsx:25]: yes

File: documentation-website/src/components/t.ts [hunk 6]
```diff
@@ -12,8 +12,15 @@ export default async(lnkey: languageKey, global?: object,): Promise<string> => {
...
   }
   const that: Window = (global ?? window) as Window;
-  const language = (that?.Navigator?.language ?? 'en')
-    .replace(/-.*$/u, '',);
+  /*const language = (
+    (that?.localStorage?.getItem('language',) || 'en')
+    ?? that?.Navigator?.language
+    ?? 'en'
+  ).replace(/-.*$/u, '',);*/
+  let language = (
```
Comment (Idrinth) [inline documentation-website/src/components/t.ts:20]: the above looks better, just replace the replace with split :)

**Review by Idrinth — CHANGES_REQUESTED** [review]

File: documentation-website/src/components/t.ts (hunk 6 above)
Comment (marikadeveloper) [reply documentation-website/src/components/t.ts]: Fixed in my latest commit (I don't really know how to fix the "conflibot / conflibot (pull_request)" failed test though 😅

**Comment by marikadeveloper** [comment]
When this is merged I have to update the languages.x keys in the en.yml and it.yml

**Pull request merged**